js/content/community.js

```diff
@@ -3700,6 +3700,98 @@ let WorkshopBrowseClass = (function(){
     return WorkshopBrowseClass;
 })();
 
+let EditGuidePageClass = (function(){
+
+    function EditGuidePageClass() {
+        this.allowMultipleLanguages();
+        this.addCustomTags();
+        this.rememberTags();
+    }
+
+    function addTag(name) {
+        let tag = `<div><input type="checkbox" name="tags[]" value="${name}" class="inputTagsFilter" checked>${name}</div>`;
+        HTML.beforeBegin("#es_add_tag", tag);
+    }
+
+    EditGuidePageClass.prototype.allowMultipleLanguages = function() {
+        document.getElementsByName("tags[]").forEach(tag => tag.type = "checkbox");
+    };
+
+    EditGuidePageClass.prototype.addCustomTags = function() {
+        let langSection = document.querySelector("#checkboxgroup_1");
+        if (!langSection) { return; }
+
+        Messenger.addMessageListener("addtag", function(name) {
+            addTag(name);
+        });
+        
+        HTML.afterEnd(langSection,
+            `<div class="tag_category_container" id="checkboxgroup_2">
+                <div class="tag_category_desc">${Localization.str.custom_tags}</div>
+                <div><a style="margin-top: 8px;" class="btn_blue_white_innerfade btn_small_thin" id="es_add_tag">
+                    <span>${Localization.str.add_tag}</span>
+                </a></div>
+            </div>`);
+
+        document.querySelector("#es_add_tag").addEventListener("click", function() {
+            ExtensionLayer.runInPageContext(`function() {
+                let Modal = ShowConfirmDialog("${Localization.str.custom_tags}", \`<div class="commentthread_entry_quotebox"><textarea placeholder="${Localization.str.enter_tag}" class="commentthread_textarea" id="es_tag" rows="1"></textarea></div>\`);
+                
+                let tag = "";
+                function done() {
+                    if (tag.trim().length === 0) { return; }
+                    tag = tag[0].toUpperCase() + tag.slice(1);
+                    Messenger.postMessage("addtag", tag);
+                }
+
+                $J("#es_tag").on("keydown paste input", function(e) {
+                    let code = e.keyCode || e.which;
+                    if (code == 13) {
+                        Modal.Dismiss();
+                        done();
```

Comment:
Why are you still trying to add the tag if the user presses Escape?
Also, isn't `#es_tag` already removed when you're executing this?

Reply:
code 13 is for enter. Yes, the element is removed, hence I save the value every time it gets changed.